builders/interoperability/xcm/xc-registration/xc-integration.md

```diff
@@ -0,0 +1,327 @@
+
+If you are registering an asset in addition to establishing a channel, please refer to the [How to Register Cross-Chain Assets](/builders/interoperability/xcm/xc-registration/assets){target=_blank} guide to learn how to generate the encoded calldata required for asset registration.
+
+The process of batching all of the transactions into one is depicted in the following diagram.
+
+![Batch XCM integration process](/images/builders/interoperability/xcm/xc-registration/xc-integration/channels-3.png)
+
+You will now use the encoded calldata outputs for opening a channel, accepting a channel, and registering an asset and insert them into the following command to send the batch proposal to democracy.
+
+You can add a `--call "INSERT_CALL"` for each call you want to batch. Replace the following values before running the command:  
+
+- `OPEN_CHANNEL_CALL` is the SCALE encoded calldata for [opening an HRMP channel](#open-an-hrmp-channel-from-moonbeam) from Moonbeam to your parachain  
+- `ACCEPT_INCOMING_CALL` is the SCALE encoded calldata for [accepting the channel request](#accept-an-hrmp-channel-on-moonbeam) from your parachain  
+- `REGISTER_ASSET_CALL` is the SCALE encoded calldata for [registering a cross-chain asset](/builders/interoperability/xcm/xc-registration/assets#register-xc-20s){target=_blank}. If you have more than one asset to be registered on Moonbeam, you can include additional registration SCALE encoded calldata with additional `--call` flags
+
+=== "Moonbeam"
+
+    ```bash
+    yarn generic-call-propose -w wss://wss.api.moonbeam.network \
+    --call "OPEN_CHANNEL_CALL" \
+    --call "ACCEPT_INCOMING_CALL" \
+    --call "REGISTER_ASSET_CALL" \
+    ```
+
+=== "Moonriver"
+
+    ```bash
+    yarn generic-call-propose -w wss://wss.api.moonriver.moonbeam.network \
+    --call "OPEN_CHANNEL_CALL" \
+    --call "ACCEPT_INCOMING_CALL" \
+    --call "REGISTER_ASSET_CALL" \
+    ```
+
+=== "Moonbase Alpha"
+
+    ```bash
+    yarn generic-call-propose -w wss://wss.api.moonbase.moonbeam.network  \
+    --call "OPEN_CHANNEL_CALL" \
+    --call "ACCEPT_INCOMING_CALL" \
+    --call "REGISTER_ASSET_CALL" \
+    ```
+
+!!! note
+    You can readapt the script for your parachain by changing the `parachain-ws-provider`.
+
+For Moonbeam, you should include `--account-priv-key YOUR_PRIVATE_KEY` and `-send-preimage-hash true --send-proposal-as democracy` (since Moonbeam uses Governance v1) if you want to send the governance proposal directly from the CLI tool. It is recommended to become familiar with the [Governance v1 process on Moonbeam-based networks](/learn/features/governance#governance-v1){target=_blank}.
```

Comment:
If this gets merged before RT240x gets rolled out to Moonbeam, I'll need to adjust this. Otherwise, I can update it in https://github.com/moonbeam-foundation/moonbeam-docs/pull/694

Reply:
As other comments